tests/activity_api_update_test.go

```diff
@@ -0,0 +1,340 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tests
+
+import (
+	"context"
+	"errors"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	activitypb "go.temporal.io/api/activity/v1"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	workflowservicepb "go.temporal.io/api/workflowservice/v1"
+	sdkclient "go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/namespace"
+	"go.temporal.io/server/common/testing/testvars"
+	"go.temporal.io/server/tests/testcore"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+)
+
+type ActivityApiUpdateClientTestSuite struct {
+	testcore.ClientFunctionalSuite
+	tv                     *testvars.TestVars
+	initialRetryInterval   time.Duration
+	scheduleToCloseTimeout time.Duration
+	startToCloseTimeout    time.Duration
+
+	activityRetryPolicy *temporal.RetryPolicy
+}
+
+func (s *ActivityApiUpdateClientTestSuite) SetupSuite() {
+	s.ClientFunctionalSuite.SetupSuite()
+	s.OverrideDynamicConfig(dynamicconfig.ActivityAPIsEnabled, true)
+	s.tv = testvars.New(s.T()).WithTaskQueue(s.TaskQueue()).WithNamespaceName(namespace.Name(s.Namespace()))
+}
+
+func (s *ActivityApiUpdateClientTestSuite) SetupTest() {
+	s.ClientFunctionalSuite.SetupTest()
+
+	s.initialRetryInterval = 10 * time.Second
+	s.scheduleToCloseTimeout = 30 * time.Second
+	s.startToCloseTimeout = 10 * time.Second
+
+	s.activityRetryPolicy = &temporal.RetryPolicy{
+		InitialInterval:    s.initialRetryInterval,
+		BackoffCoefficient: 1,
+		MaximumInterval:    10 * time.Second,
+	}
+}
+
+func TestActivityApiUpdateClientTestSuite(t *testing.T) {
+	s := new(ActivityApiUpdateClientTestSuite)
+	suite.Run(t, s)
+}
+
+func (s *ActivityApiUpdateClientTestSuite) waitForChan(ctx context.Context, ch chan struct{}) {
+	s.T().Helper()
+	select {
+	case <-ch:
+	case <-ctx.Done():
+		s.FailNow("context timeout")
+	}
+}
+
+type (
+	ActivityFunctions func() (string, error)
+	WorkflowFunction  func(context2 workflow.Context) (string, error)
+)
+
+func (s *ActivityApiUpdateClientTestSuite) makeWorkflowFunc(activityFunction ActivityFunctions) WorkflowFunction {
+	return func(ctx workflow.Context) (string, error) {
+
+		var ret string
+		err := workflow.ExecuteActivity(workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+			ActivityID:             "activity-id",
+			DisableEagerExecution:  true,
+			StartToCloseTimeout:    s.startToCloseTimeout,
+			ScheduleToCloseTimeout: s.scheduleToCloseTimeout,
```

Comment:
I wouldn't set this. It doesn't give you much. You have a context deadline in the test already that will prevent it from running too long.